Elaborate on the behavior of extends in classes (#16293)
mdn/content: files/en-us/web/javascript/reference/classes/extends/index.md

The `extends` keyword can be used to subclass custom classes as well as
built-in objects.

> Any constructor that can be called with `new` (i.e. it has the `prototype` property) can be the candidate for the parent class.

Reviewer: ```suggestion
Any constructor that can be called with `new` (that is, it has the `prototype` property) can be the candidate for the parent class.
```

```js
function OldStyleClass() {
  this.someProperty = 1;
}
OldStyleClass.prototype.someMethod = function () {};

class ChildClass extends OldStyleClass {}

class ModernClass {
  someProperty = 1;
  someMethod() {}
}

class AnotherChildClass extends ModernClass {}
```

The `.prototype` of the `ParentClass` must be an {{jsxref("Object")}} or {{jsxref("null")}}. 

```js
function ParentClass() {}
ParentClass.prototype = 3;

class ChildClass extends ParentClass {}
// Uncaught TypeError: Class extends value does not have valid prototype property 3
```

> > **Note:** You would rarely worry about this in practice, because a non-object `prototype` doesn't behave as it should anyways.

Reviewer: ```suggestion
> **Note:** You would rarely worry about this in practice, because a non-object `prototype` doesn't behave as it should anyway.
```
